program/tests/helpers/mint_builder.rs

```diff
@@ -0,0 +1,226 @@
+use {
+    crate::helpers::common::{init_mollusk, setup_mint},
+    mollusk_svm::{result::Check, Mollusk},
+    solana_account::Account,
+    solana_program::system_program,
+    solana_pubkey::Pubkey,
+    spl_token_wrap::{
+        get_wrapped_mint_address, get_wrapped_mint_backpointer_address, instruction::create_mint,
+    },
+};
+
+pub struct CreateMintResult {
+    pub unwrapped_mint: KeyedAccount,
+    pub wrapped_mint: KeyedAccount,
+    pub wrapped_backpointer: KeyedAccount,
+}
+
+#[derive(Debug, Clone)]
+#[cfg(test)]
+pub struct KeyedAccount {
+    pub key: Pubkey,
+    pub account: Account,
+}
+
+#[cfg(test)]
+impl KeyedAccount {
+    #[cfg(test)]
+    pub fn pair(&self) -> (Pubkey, Account) {
+        (self.key, self.account.clone())
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum TokenProgram {
+    SplToken,
+    Token2022,
+}
+
+impl TokenProgram {
+    pub fn id(&self) -> Pubkey {
+        match self {
+            TokenProgram::SplToken => spl_token::id(),
+            TokenProgram::Token2022 => spl_token_2022::id(),
+        }
+    }
+
+    pub fn keyed_account(&self) -> (Pubkey, Account) {
+        match self {
+            TokenProgram::SplToken => mollusk_svm_programs_token::token::keyed_account(),
+            TokenProgram::Token2022 => mollusk_svm_programs_token::token2022::keyed_account(),
+        }
+    }
+}
+
+pub struct MintBuilder<'a> {
+    mollusk: Mollusk,
+    wrapped_token_program: TokenProgram,
+    wrapped_token_program_addr: Option<Pubkey>,
+    unwrapped_mint_addr: Option<Pubkey>,
+    unwrapped_mint_account: Option<Account>,
+    unwrapped_token_program: TokenProgram,
+    wrapped_mint_addr: Option<Pubkey>,
+    wrapped_mint_account: Option<Account>,
+    backpointer_addr: Option<Pubkey>,
+    backpointer_account: Option<Account>,
+    idempotent: bool,
+    checks: Vec<Check<'a>>,
+}
+
+impl<'a> Default for MintBuilder<'a> {
+    fn default() -> Self {
+        Self {
+            mollusk: init_mollusk(),
+            wrapped_token_program: TokenProgram::Token2022,
+            wrapped_token_program_addr: None,
```

Comment:
Is this needed? `wrapped_token_program.id()` should give the info, no?

Reply:
If none is passed, it will indeed grab from there. It was added as a test param [for this test](https://github.com/solana-program/token-wrap/pull/10/files#diff-50e76d72138fd69729c6e469dc9201b57b77542f9c631bbe8f5077821dd473b3R159-R163) that tests if someone passes the wrong address for the token program. But maybe not needed?

Reply:
Ah sorry, I forgot to remove this comment after going through the tests -- yep looks good!